packages/contracts-bedrock/src/L1/DataAvailabilityChallenge.sol

```diff
@@ -0,0 +1,436 @@
+    }
+
+    /// @notice Checks if the current block is within the resolve window for a given challenge start block number.
+    /// @param challengeStartBlockNumber The block number at which the challenge was initiated.
+    /// @return True if the current block is within the resolve window, false otherwise.
+    function _isInResolveWindow(uint256 challengeStartBlockNumber) internal view returns (bool) {
+        return block.number <= challengeStartBlockNumber + resolveWindow;
+    }
+
+    /// @notice Returns a challenge for the given block number and commitment.
+    /// @dev Unlike with a public `challenges` mapping, we can return a Challenge struct instead of tuple.
+    /// @param challengedBlockNumber The block number at which the commitment was made.
+    /// @param challengedCommitment The commitment that is being challenged.
+    /// @return The challenge struct.
+    function getChallenge(
+        uint256 challengedBlockNumber,
+        bytes calldata challengedCommitment
+    )
+        public
+        view
+        returns (Challenge memory)
+    {
+        return challenges[challengedBlockNumber][challengedCommitment];
+    }
+
+    /// @notice Returns the status of a challenge for a given challenged block number and challenged commitment.
+    /// @param challengedBlockNumber The block number at which the commitment was made.
+    /// @param challengedCommitment The commitment that is being challenged.
+    /// @return The status of the challenge.
+    function getChallengeStatus(
+        uint256 challengedBlockNumber,
+        bytes calldata challengedCommitment
+    )
+        public
+        view
+        returns (ChallengeStatus)
+    {
+        Challenge memory _challenge = challenges[challengedBlockNumber][challengedCommitment];
+        // if the address is 0, the challenge is uninitialized
+        if (_challenge.challenger == address(0)) return ChallengeStatus.Uninitialized;
+
+        // if the challenge has a resolved block, it is resolved
+        if (_challenge.resolvedBlock != 0) return ChallengeStatus.Resolved;
+
+        // if the challenge's start block is in the resolve window, it is active
+        if (_isInResolveWindow(_challenge.startBlock)) return ChallengeStatus.Active;
+
+        // if the challenge's start block is not in the resolve window, it is expired
+        return ChallengeStatus.Expired;
+    }
+
+    /// @notice Validate that a given commitment has a known type and the expected length for this type.
+    /// @dev The type of a commitment is stored in its first byte.
+    ///      The function reverts with `UnknownCommitmentType` if the type is not known and
+    ///      with `InvalidCommitmentLength` if the commitment has an unexpected length.
+    /// @param commitment The commitment for which to check the type.
+    /// @return The commitment type of the given commitment.
+    function _validateCommitment(bytes calldata commitment) internal pure returns (CommitmentType) {
```

Comment:
It could be useful to make this public and also directly test it including the revert case

Reply:
It is also strange that a validation function returns the type, seems like its coupling some functionality but not that big of a deal, more of a naming nit than anything else

Reply:
This coupled abstraction is clear in the case where the return value isn't used in `challenge()`

Reply:
agreed, updated